client/components/AddSponsor.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+
+interface AddSponsorProps {
+  eventId: string;
+  chapterId: string;
+}
+
+interface SponsorData {
+  name?: string;
+  website?: string;
+  type?: string;
+}
+
+const AddSponsor: React.FC<AddSponsorProps> = ({ eventId, chapterId }) => {
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
+  const [values, setValues] = React.useState<SponsorData>({ type: 'FOOD' });
+  const [success, setSuccess] = React.useState<boolean>(false);
+
+  function handleChange(e) {
+    const { value, name } = e.target;
+    setValues({
+      ...values,
+      [name]: value,
+    });
+  }
+
+  async function handleSubmit(e) {
+    /* Block double submits */
+    if (isSubmitting) {
+      return false;
+    }
+
+    e.preventDefault();
+    setSubmitting(true);
+
+    try {
+      const data = JSON.stringify(values);
+      await fetch(`/${chapterId}/events/${eventId}/sponsors`, {
```

Comment:
add a TODO in a comment to indicate this route doesn't exist yet

Reply:
added